lib/update.js

```diff
@@ -85,6 +85,12 @@ function update(updates, object, ...args) {
 
   const resolvedUpdates = resolveUpdates(updates, defaultedObject)
 
+  Object.entries(resolvedUpdates).forEach(([key, value]) => {
```

Comment:
This probably warrants a comment explaining what this is doing at a high level

Reply:
Not a bad idea. Done!